File: pypit/oop.py
class Citizen(object):

    def __init__(self, name, country):
        self.name = name
        self.country = country

    def print_details(self):
        print('Citizen %s from %s' % (self.name, self.country))


class Winner(Citizen):

    def __init__(self, name, country, category, year):
        super(Winner, self).__init__(name, country)
        self.category = category
        self.year = year

    def print_details(self):
        print('Academy Award winner %s from %s, category %s, year %s'
              % (self.name, self.country, self.category, str(self.year)))


c = Citizen('Adele', 'England')
c.print_details()

w = Winner('Adele', 'England', 'Best Original Song', 2013)
w.print_details()

File: sandpit/oop.js
"use strict";

let Citizen = {
	setCitizen: function (name, country) {
		this.name = name;
		this.country = country;
		return this;
	},
	printDetails: function () {
		console.log(`Citizen ${this.name} from ${this.country}`);
	}
};



let Winner = Object.create(Citizen);

Winner.setWinner = function (name, country, category, year) {
	this.setCitizen(name, country);
	this.category = category;
	this.year = year;
	return this;
};

Winner.printDetails = function () {
	console.log(`Academy Award winner ${this.name} from ${this.country}, category ${this.category}, year ${this.year}`);
};

let adele = Object.create(Winner).setWinner('Adele', 'England', 'Best Original Song', 2013);

adele.printDetails();